pkg/sql/conn_executor.go

```diff
@@ -1466,6 +1466,10 @@ func (ex *connExecutor) resetExtraTxnState(ctx context.Context, ev txnEvent) err
 		p.decRef(ctx, &ex.extraTxnState.prepStmtsNamespaceMemAcc, name)
 		delete(ex.extraTxnState.prepStmtsNamespace.portals, name)
 	}
+	for name, p := range ex.extraTxnState.prepStmtsNamespaceAtTxnRewindPos.portals {
```

Comment:
i noticed this comment on `prepStmtsNamespaceAtTxnRewindPos`:
> Prepared statements are shared between the two collections, but these collections are periodically reconciled.

i wonder if we lose track of a reference when that happens?